precompiles/avsTask/task.go

```diff
@@ -0,0 +1,132 @@
+package task
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	taskKeeper "github.com/ExocoreNetwork/exocore/x/avs_task/keeper"
+	tasktype "github.com/ExocoreNetwork/exocore/x/avs_task/types"
+	"github.com/cometbft/cometbft/libs/log"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	cmn "github.com/evmos/evmos/v14/precompiles/common"
+)
+
+var _ vm.PrecompiledContract = &Precompile{}
+
+// Embed abi json file to the executable binary. Needed when importing as dependency.
+//
+//go:embed abi.json
+var f embed.FS
+
+// Precompile defines the precompiled contract for task.
+type Precompile struct {
+	cmn.Precompile
+	taskKeeper taskKeeper.Keeper
+	avsKeeper  tasktype.AvsKeeperMock
+}
+
+// NewPrecompile creates a new task Precompile instance as a
+// PrecompiledContract interface.
+func NewPrecompile(
+	authzKeeper authzkeeper.Keeper,
+	taskKeeper taskKeeper.Keeper,
+	avsKeeper tasktype.AvsKeeperMock,
```

Comment:
Please change all mock to real implement type

Reply:
Fixed